packages/emotion-theming/src/theme-provider.js

```diff
@@ -1,19 +1,27 @@
+// @flow
 // adapted from styled-components' ThemeProvider
 // https://github.com/styled-components/styled-components/blob/4503cab5b86aa9ef8314c5baa360a2fbb4812485/src/models/ThemeProvider.js
 
-import React, { Component } from 'react'
+import { Component, Children } from 'react'
 import createBroadcast from './create-broadcast'
-import { channel, contextTypes } from './utils'
+import { channel, contextTypes, type Theme } from './utils'
 
 const isPlainObject = test =>
   Object.prototype.toString.call(test) === '[object Object]'
 
-class ThemeProvider extends Component {
+type Props = {
+  theme: Theme
+}
+
+class ThemeProvider extends Component<Props> {
+  getTheme: ThemeProvider.prototype.getTheme
```

Comment:
had no idea a type can be specified like this, is there reason why it's defined like this and why it's defined at all? i mean - methods should be inferrable from the class definition, right?

Reply:
Flow complains without it because of the bind in the constructor https://flow.org/try/#0PQKgBAAgZgNg9gdzCYAoVBjGBDAzrsAFQAsBLAOwHMwBvVMMDOc3AFwCcBXDVudgCgCUtemFZlcAOlxwAtgFMAsvPFwAJmAC8YidLlKVxdZIBGFNf3Glcg0QF9RMhctUXhdBvdR2gA

Reply:
wouldnt `transform-class-properties` fix this?